api/swagger.yml

```diff
@@ -1,507 +1,795 @@
-      default_branch:
-        example: "master"
-        type: string
-      storage_namespace:
-        type: string
-        description: "Filesystem URI to store the underlying data in (e.g. 's3://my-bucket/some/path/')"
-
-  merge_result:
-    type: object
-    properties:
-      summary:
-        type: object
-        properties:
-          added:
-            type: integer
-          removed:
-            type: integer
-          changed:
-            type: integer
-          conflict:
-            type: integer
-      reference:
-        type: string
-
-  repository_creation:
-    type: object
-    required:
-      - name
-      - storage_namespace
-    properties:
-      name:
-        type: string
-        pattern: '^[a-z0-9][a-z0-9-]{2,62}$'
-      storage_namespace:
-        type: string
-        description: "Filesystem URI to store the underlying data in (e.g. 's3://my-bucket/some/path/')"
-        example: "s3://example-bucket/"
-        pattern: '^(s3|gs|https?|mem|local|transient)://.*$'
-      default_branch:
-        example: "master"
-        type: string
+        minimum: -1
+        maximum: 1000
+        default: 100
 
-  object_stats:
-    type: object
-    properties:
-      path:
-        type: string
-      physical_address:
-        type: string
-      checksum:
-        type: string
-      mtime:
-        type: integer
-        format: int64
-      size_bytes:
-        type: integer
-        format: int64
-        x-nullable: true
-      path_type:
-        type: string
-        enum: [ common_prefix, object ]
+  responses:
+    Unauthorized:
+      description: Unauthorized
+      content:
+        application/json:
+          schema:
+            $ref: "#/components/schemas/Error"
+    ServerError:
+      description: Internal Server Error
+      content:
+        application/json:
+          schema:
+            $ref: "#/components/schemas/Error"
+    NotFound:
+      description: Resource Not Found
+      content:
+        application/json:
+          schema:
+            $ref: "#/components/schemas/Error"
+    ValidationError:
+      description: Validation Error
+      content:
+        application/json:
+          schema:
+            $ref: "#/components/schemas/Error"
+
+
+  schemas:
+
+    Pagination:
+      type: object
+      required:
+        - has_more
+        - max_per_page
+        - results
+        - next_offset
+      properties:
+        has_more:
+          type: boolean
+        next_offset:
+          type: string
+        results:
+          type: integer
+          minimum: 0
+        max_per_page:
+          type: integer
+          minimum: 0
+
+    Repository:
+      type: object
+      required:
+        - creation_date
+        - id
+        - default_branch
+        - storage_namespace
+      properties:
+        id:
+          type: string
+        creation_date:
+          type: integer
+          format: int64
+        default_branch:
+          example: "master"
```

Comment:
Good chance to go for `example: "main"` or `example: "trunk"` (your call).  We know we should, we know we shall, we're touching it now...